unificontrol/pinned_requests.py

```diff
@@ -55,8 +56,11 @@ def __init__(self, cert, ca_cert=None, **kwargs):
         for self-signed certs)"""
         if ca_cert is None:
             ca_cert = cert
-        # The temporary file will be cleaned up when this adaptor is garbage collected
-        self._ca_cert_temp = tempfile.NamedTemporaryFile(mode="w", suffix=".pem")
+        # The temporary file will be cleaned up when this adaptor is garbage collected, except
+        # on Windows where this behaviour prevents the ssl library from opening the file
+        self._ca_cert_temp = tempfile.NamedTemporaryFile(mode="w",
+                                                         suffix=".pem",
+                                                         delete=(os.name != 'nt'))
```

Comment:
Is `os.name != 'nt'` sufficient here? Does this cover all types of Windows?

Reply:
It does. os.name is, I think, the bluntest way to determine the OS - since 3.4 its possible values are "posix", "nt", or "java". This construct clears the delete flag on any version of windows, but leaves the default behaviour (`delete=True`) otherwise... I'll admit I have no idea how NamedTemporaryFile behaves in relation to this issue on a Java based system.